Entities/AtomGroup.cpp

```diff
@@ -101,29 +105,31 @@ int AtomGroup::Create(const AtomGroup &reference)
     m_Resolution = reference.m_Resolution;
     m_Depth = reference.m_Depth;
 
-    int subID = 0;
-    m_SubGroups.clear();
-    for (list<Atom *>::const_iterator itr = reference.m_Atoms.begin(); itr != reference.m_Atoms.end(); ++itr)
-    {
-        Atom *pAtomCopy = new Atom(**itr);
-		pAtomCopy->SetIgnoreMOIDsByGroup(&m_IgnoreMOIDs);
+	m_SubGroups.clear();
+	for (list<Atom *>::const_iterator itr = reference.m_Atoms.begin(); itr != reference.m_Atoms.end(); ++itr)
+	{
+		if (!onlyCopyOwnerAtoms || (*itr)->GetSubID() == 0)
+		{
+			Atom *pAtomCopy = new Atom(**itr);
+			pAtomCopy->SetIgnoreMOIDsByGroup(&m_IgnoreMOIDs);
 
-        m_Atoms.push_back(pAtomCopy);
+			m_Atoms.push_back(pAtomCopy);
 
-		// TODO: OPTIMIZE THE SHIT OUT OF THIS!!! TERRIBLE to have a find in every atom creation for a group!
-        // Add to the appropriate spot in the subgroup map
-        int subID = pAtomCopy->GetSubID();
-        if (subID != 0)
-        {
-            // Try to find the group
-            map<int, list<Atom *> >::iterator subItr = m_SubGroups.find(subID);
-            // No atom added to that group yet, so it doesn't exist, so make it
-            if (subItr == m_SubGroups.end())
-                subItr = (m_SubGroups.insert(pair<int, list<Atom *> >(subID, list<Atom *>()))).first;
-            // Add Atom to the list of that group
-            subItr->second.push_back(pAtomCopy);
-        }
-    }
+			// Add to the appropriate spot in the subgroup map
+			long int subID = pAtomCopy->GetSubID();
+			if (subID != 0)
+			{
+				// Try to find the group
+				// No atom added to that group yet, so it doesn't exist, so make it
+				if (m_SubGroups.find(subID) == m_SubGroups.end())
+				{
+					m_SubGroups.insert(pair<long int, list<Atom *> >(subID, list<Atom *>())).first;
```

Comment:
I'm removing this `.first` and the next one, since they're junk from when were were using iterators here.